demo_scripts/AKSDeployment/README.md

```diff
@@ -0,0 +1,314 @@
+```
+
+Get Public IP ID
+
+```
+export PUBLIC_IP_ID=$(az network public-ip list --query "[?ipAddress!=null]|[?contains(ipAddress, '$IP_ADDRESS')].[id]" --output tsv)
+```
+
+Set Custom DNS Name by running the following:
+'az network public-ip update --ids $PUBLIC_IP_ID --dns-name $CUSTOM_DOMAIN_NAME'
+
+```
+az network public-ip update --ids $PUBLIC_IP_ID --dns-name $CUSTOM_DOMAIN_NAME
+```
+
+Check custom domain to see application running - 
+'az network public-ip show --ids $PUBLIC_IP_ID --query "[dnsSettings.fqdn]" --output tsv'
+```
+az network public-ip show --ids $PUBLIC_IP_ID --query "[dnsSettings.fqdn]" --output tsv
+```
+
+Store Custom Domain by running the following:
+'export FQDN=$(az network public-ip show --ids $PUBLIC_IP_ID --query "[dnsSettings.fqdn]" --output tsv)'
+```
+export FQDN=$(az network public-ip show --ids $PUBLIC_IP_ID --query "[dnsSettings.fqdn]" --output tsv)
+```
+
+# Part 3 Install SSL
+Create namespace for cert manager by running the following:
+'kubectl create namespace cert-manager'
+```
+kubectl create namespace cert-manager
+```
+
+Apply Cert-Manager by running the following:
+
+'kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.0/cert-manager.crds.yaml'
+
+```
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.0/cert-manager.crds.yaml
+```
+
+Label Namespace by running the following:
+
+'kubectl label namespace cert-manager certmanager.k8s.io/disable-validation=true'
+```
+kubectl label namespace cert-manager certmanager.k8s.io/disable-validation=true
+```
+
+Add helm to Jetstack Repo by running the following:
+'helm repo add jetstack https://charts.jetstack.io'
+```
+helm repo add jetstack https://charts.jetstack.io
+```
+Update Repo by running the following:
+'helm repo update'
+```
+helm repo update
+```
+
+Install Cert-Manager via helm by running the following:
+'helm install cert-manager jetstack/cert-manager --namespace cert-manager --version v1.7.0'
+```
+helm install cert-manager jetstack/cert-manager --namespace cert-manager --version v1.7.0
+```
+
+Apply CertIssuer YAML File by running the following:
+'envsubst < cluster-issuer-prod.yaml | kubectl apply -f -'
+```
+envsubst < cluster-issuer-prod.yaml | kubectl apply -f -
+```
+
+Apply Updated YAML file with SSL Certby running the following:
+envsubst < azure-vote-agic-ssl.yml | kubectl apply -f -
+```
+envsubst < azure-vote-agic-ssl.yml | kubectl apply -f -
+```
+
+Check to make sure Ingress is working by running:
+'for run in {1..3}; do kubectl get ingress; sleep 5; done'
+
+Note - We are running this multiple times to show the Lets Encrypt Service validating domain ownership
+```
+for run in {1..3}; do kubectl get ingress; sleep 5; done
+```
+
+Check SSL Certificate by running the following command:
+'kubectl get certificate'
+```
+kubectl get certificate
```

Comment:
For me this returned a false state in the ready column for around 15 minutes. We should block on this for a period of time until either a max time is exceeded (failure) or the cert is ready. Otherwise the next step fails.